File: union-find/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        store = set(nums)
        for n in nums:
            count = 0
            if (n-1) not in store:
                count=0
                while (n+count) in store:
                    count+=1
                res = max(res, count)
        return res

File: array/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> store = new HashSet<>();
        for (int num: nums){
            store.add(num);
        }

        int res = 0;

        for (int n: nums){
            if(!store.contains(n-1)){
                int currentNum = n;
                int currLen = 1;

                while (store.contains(n+ 1)){
                    currLen++;
                    currentNum++;
                }
                res = Math.max(currLen, res); 

            }
        }
        return res;
        
    }
}